File: Python/beginner/1118.py
# https://judge.beecrowd.com/pt/problems/view/1118
# Várias Notas Com Validação

def get_grade():
    while True:
        try:
            grade = float(input())
            if 0 <= grade <= 10:
                return grade
            else:
                print("nota invalida")
        except ValueError:
            print("nota invalida")

def get_repeat():
    while True:
        print("novo calculo (1-sim 2-nao)")
        decision = input()
        if decision == "1":
            return True
        elif decision == "2":
            return False
            

def main():
    while True:
        # Solicitando notas
        grade1 = get_grade()
        grade2 = get_grade()
        # Calculando e apresentando a média
        media = (grade1 + grade2) / 2
        print(f"media = {media:.2f}")
        
        should_repeat = get_repeat()
        if not should_repeat:
            break

if __name__ == "__main__":
    main()
    
# By Fernando Serra
# https://github.com/fernandosserra

File: Haskell/beginner/1118.hs
-- https://judge.beecrowd.com.br/judge/en/problems/view/1118
-- Várias notas com validação


-- Importing printf
import Text.Printf

-- Função para obter uma nota válida
getGrade :: IO Double
getGrade = do
    grade <- readLn
    if grade < 0 || grade > 10
        then do
            printf "nota invalida\n"
            getGrade
        else return grade

-- Função para verificar se um novo cálculo é necessário
getRepeat :: IO Bool
getRepeat = do
    printf "novo calculo (1-sim 2-nao)\n"
    decision <- readLn
    case decision of
        1 -> return True
        2 -> return False
        _ -> do
            getRepeat

-- Bloco principal
main :: IO ()
main = loop
  where
    loop = do
        -- Solicitando notas
        grade1 <- getGrade
        grade2 <- getGrade
        -- Calculando e apresentando a média
        let media = (grade1 + grade2) / 2
        printf "media = %.2f\n" media
        shouldRepeat <- getRepeat
        if shouldRepeat
            then loop
            else return ()


-- By Fernando Serra
-- In this case i've used a little help fro ChatGPT to learn Haskell and understand some indentations.
-- https://github.com/fernandosserra
